cmd/notation/plugin.go

```diff
@@ -62,3 +108,169 @@ func listPlugins(command *cobra.Command) error {
 	}
 	return tw.Flush()
 }
+
+func installPlugin(command *cobra.Command, args []string, force bool) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin package")
+	}
+
+	pluginPath := args[0]
+
+	var pluginBinary string
+	// if plugin contains a file path split and select the last element in the array
+	if strings.Contains(pluginPath, "/") {
+		pluginBinary = strings.Split(pluginPath, "/")[len(strings.Split(pluginPath, "/"))-1]
+	} else {
+		pluginBinary = pluginPath
+	}
+
+	// get plugin metadata
+	cmd := exec.Command("./"+pluginPath, "get-plugin-metadata")
+
+	output, err := cmd.Output()
+	if err != nil {
+		panic(err)
+	}
+
+	var newPlugin map[string]interface{}
+	err = json.Unmarshal([]byte(output), &newPlugin)
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
```

Comment:
We can reuse the [GetMetadata](https://pkg.go.dev/github.com/notaryproject/notation-go/plugin#CLIPlugin.GetMetadata) to get the name and version of plugin.

Reply:
I'm using the exec func to run [GetMetadata](https://github.com/notaryproject/notation/blob/26187094a03643e0b283d40f34e39cf599df75dd/cmd/notation/plugin.go#L128) then using Unmarshal to convert the JSON results to a Go object. If there is a better way to do that, please lmk. :)

Reply:
[Changes ](https://github.com/notaryproject/notation/pull/617/commits/e4d7ef5f0b4d53f2e1a218350742fab9f09c21ff) made to use the `CLIPlugin`.